File: spring-crud-api/src/main/java/com/spring/crud/app/example/api/error/ErrorDetail.java
package com.spring.crud.app.example.api.error;

public class ErrorDetail {

    private String title;
    private int status;
    private String detail;
    private long timestamp;
    private String developerMessage;

    public String getTitle() {
        return title;
    }

    void setTitle(String title) {
        this.title = title;
    }

    public int getStatus() {
        return status;
    }

    void setStatus(int status) {
        this.status = status;
    }

    public String getDetail() {
        return detail;
    }

    void setDetail(String detail) {
        this.detail = detail;
    }

    public long getTimestamp() {
        return timestamp;
    }

    void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getDeveloperMessage() {
        return developerMessage;
    }

    void setDeveloperMessage(String developerMessage) {
        this.developerMessage = developerMessage;
    }

    public static final class Builder {
        private String title;
        private int status;
        private String detail;
        private long timestamp;
        private String developerMessage;

        private Builder() {
        }

        public static Builder newBuilder() {
            return new Builder();
        }

        public Builder title(String title) {
            this.title = title;
            return this;
        }

        public Builder status(int status) {
            this.status = status;
            return this;
        }

        public Builder detail(String detail) {
            this.detail = detail;
            return this;
        }

        public Builder timestamp(long timestamp) {
            this.timestamp = timestamp;
            return this;
        }

        public Builder developerMessage(String developerMessage) {
            this.developerMessage = developerMessage;
            return this;
        }

        public ErrorDetail build() {
            ErrorDetail errorDetail = new ErrorDetail();
            errorDetail.setTitle(title);
            errorDetail.setStatus(status);
            errorDetail.setDetail(detail);
            errorDetail.setTimestamp(timestamp);
            errorDetail.setDeveloperMessage(developerMessage);
            return errorDetail;
        }
    }
}

File: sales-catalog-app/src/main/kotlin/com/generoso/salescatalog/exception/error/ErrorDetail.kt
package com.generoso.salescatalog.exception.error

import java.time.LocalDateTime

open class ErrorDetail {
    var status: Int? = null
    var detail: String? = null
    var dateTime: LocalDateTime? = null

    companion object {
        fun builder(): Builder {
            return Builder()
        }
    }

    class Builder {
        private var status: Int? = null
        private var detail: String? = null
        private var dateTime: LocalDateTime? = null

        fun status(status: Int): Builder {
            this.status = status
            return this
        }

        fun detail(detail: String?): Builder {
            this.detail = detail
            return this
        }

        fun dateTime(dateTime: LocalDateTime): Builder {
            this.dateTime = dateTime
            return this
        }

        fun build(): ErrorDetail {
            val errorDetail = ErrorDetail()
            errorDetail.status = status
            errorDetail.detail = detail
            errorDetail.dateTime = dateTime
            return errorDetail
        }
    }
}
